src/vivarium_gates_nutrition_optimization_child/components/fertility.py

```diff
@@ -63,10 +63,12 @@ def on_time_step(self, event: Event) -> None:
             The event that triggered the function call.
         """
         birth_records = self.birth_records
+        birth_records["birth_date"] = pd.to_datetime(birth_records["birth_date"])
```

Comment:
Nit: i think you should put this reformatting at the end of `_get_birth_records` instead of here, and then it only needs to be executed once instead of every timestep.